packages/app/components/DatePicker.tsx

```diff
@@ -0,0 +1,170 @@
+  fromDate?: Date;
+}
+
+export function DatePicker({
+  dateTime,
+  setDateTime,
+  className,
+  timeCaption,
+  fromDate,
+}: IDatePicker) {
+  const [currentDate, setCurrentDate] = useState<Date>(new Date(dateTime));
+  const [hours, setHours] = useState(format(currentDate, "HH"));
+  const [minutes, setMinutes] = useState(format(currentDate, "mm"));
+
+  const [referenceElement, setReferenceElement] = useState<
+    HTMLButtonElement | undefined
+  >();
+  const [popperElement, setPopperElement] = useState<
+    HTMLDivElement | undefined
+  >();
+
+  const { styles, attributes } = usePopper(referenceElement, popperElement, {
+    placement: "bottom",
+  });
+
+  const formattedDate = () => {
+    if (isToday(dateTime)) return format(dateTime, "'Today at' HH:mm");
+    if (isTomorrow(dateTime)) return format(dateTime, "'Tomorrow at' HH:mm");
+
+    return format(dateTime, "dd MMM Y @ HH:mm");
+  };
+
+  useEffect(() => {
+    const newDate = new Date(dateTime);
+    setCurrentDate(newDate);
+    setHours(format(newDate, "HH"));
+    setMinutes(format(newDate, "mm"));
+  }, [dateTime]);
+
+  return (
+    <Popover>
+      {({ open }) => (
+        <>
+          <Popover.Button
+            ref={setReferenceElement as Ref<HTMLButtonElement>}
+            className={twMerge(
+              "flex justify-between items-center focus:border-0",
+              className
+            )} // text-em-low if default dates
+          >
+            <span>{formattedDate()}</span>
+            <Icon name="caret-down" className="mr-2 h-4 w-4 text-black" />
+          </Popover.Button>
+          {open && (
+            <>
+              <div className="fixed inset-0 bg-black/30" aria-hidden="true" />
+              <Popover.Panel
+                ref={setPopperElement as Ref<HTMLDivElement>}
+                style={styles.popper}
+                {...attributes.popper}
+                className="w-fit h-fit m-auto max-md:!fixed max-md:!inset-0 max-md:!flex max-md:!items-center max-md:!justify-center max-md:!transform" /* This styles will make it hover on mobile */
+              >
+                {({ close }) => (
+                  <div className="flex flex-col space-y-2 bg-white divide-y divide-surface-50 rounded-2xl border border-surface-50">
+                    <Calendar
+                      mode="single"
+                      selected={currentDate}
+                      defaultMonth={currentDate}
+                      onSelect={(newDate: Date | undefined) => {
+                        if (!newDate) return;
+                        newDate.setHours(currentDate.getHours());
+                        newDate.setMinutes(currentDate.getMinutes());
+                        setCurrentDate(newDate);
+                      }}
+                      initialFocus
+                      className=""
+                      fromDate={fromDate || new Date()}
+                    />
+                    <div className="flex flex-col space-y-2 p-4 pt-2">
```

Comment:
```suggestion
                    <div className="flex flex-col space-y-2 p-4">
```

We have two paddings here, I think pt-2 is not doing anything we can remove it